### ISSUE TITLE
Accept: application/json

### ISSUE THREAD

**Issue by patrickgardella** [ewoutkramer/fhir-net-api#542]
Good afternoon!

I'm trying to use the CMS Blue Button API with the fhir-net-api, and keep getting 406 errors from them. From what I can determine, fhir-net-api will send either "application/fhir+json" or "application/fhir+xml" as the content type (per the [HL7 specs](https://www.hl7.org/fhir/json.html)). The Blue Button API only accepts "application/json" content type. ([metadata statement](https://sandbox.bluebutton.cms.gov/v1/fhir/metadata)) 

Is there a way to force the Accept header to the desired MIME type?

I've tried this:
```
            var client = new FhirClient("https://sandbox.bluebutton.cms.gov/v1/fhir");
            client.PreferredFormat = ResourceFormat.Json;

           client.OnBeforeRequest += (object sender, BeforeRequestEventArgs e) =>
            {
                // Replace with a valid bearer token for this server
                e.RawRequest.Headers.Add("Authorization", "Bearer " + accessToken);
                e.RawRequest.Headers.Remove("Accept");
                e.RawRequest.Headers.Add("Accept", "application-json");
            };
```
This fails because the client adds the header after this call. (Or at least I think it does!)

Thanks! 
Patrick

**Comment by brianpos** [comment]
Then i would say that the blue button API is not conformant there.
The correct thing is to have the preferred format as you’ve done, and get them to fix their bug.

Regards
Brian

**Comment by ewoutkramer** [comment]
I agree with @brianpos, but still, I think it makes sense to not manipulate headers anymore after OnBeforeRequest - this is a change we could make anyway?

**Comment by brianpos** [comment]
I read that as the server should do the right thing, bit forgiving of clients. Like JavaScript.
